### PR TITLE
experiment skeleton

### PR DESCRIPTION
**Pull request by yongjunshin**
new parameters
1. experiment repeat 
2. manual seed

refactoring
1. main파일 (environment_model_generation.py) 에서 argument parsing과 필요한 함수 호출 외의 일 최대한 안하도록 skeleton화

### CODE REVIEW

File: environment_model_generation.py [hunk 1]
```diff
@@ -84,7 +79,16 @@ def mode_selection(args):
         episode_length = history_length
     else:
         episode_length = args.episode_length
-
+manual_seed = args.manual_seed
+if manual_seed is not None:
```
Comment (donghwan-shin) [inline environment_model_generation.py:83]: `manual_seed`가 없는 경우에도 내부적으로 사용된 seed값을 결과와 함께 출력할 수 있어야 나중에 해당 실험내용을 reproduce할 수 있음! 어떤 seed값이 사용되었는지 리턴받는 함수가 있다면 그걸 쓰고, 아니면 애초에 time.time() 같은걸 사용해서 특정 seed값을 생성해서 그걸 사용하도록 해야 할듯

File: environment_model_generation.py [hunk 2]
```diff
@@ -103,119 +107,64 @@ def mode_selection(args):
...
+    print("--data size:", [raw_df.shape for raw_df in raw_dfs])
+
+    print("Step 2: Data normalization")
+    noramlized_nparrays, scaler = normalize_dataframes_to_nparrays(raw_dfs)
+
+    print("Step 3: Build train/test/validation dataset")
+    # Build train/test/validation dataset
+    train_dataloaders, test_dataloaders, validation_dataloaders = build_train_test_validation_dataset(noramlized_nparrays, mode, history_length, None, batch_size, device)
+
+    print("--train dataset shape:", [str(loader.dataset.x.shape) +'->' + str(loader.dataset.y.shape) for loader in train_dataloaders])
```
Comment (donghwan-shin) [inline environment_model_generation.py:139]: 작은 팁인데, python 3.6 이후부턴가 print(f'') 형태를 사용할수 있는데 엄청 편해! 이 [링크](https://www.datacamp.com/community/tutorials/f-string-formatting-in-python?utm_source=adwords_ppc&utm_campaignid=898687156&utm_adgroupid=48947256715&utm_device=c&utm_keyword=&utm_matchtype=b&utm_network=g&utm_adpostion=&utm_creative=332602034352&utm_targetid=dsa-429603003980&utm_loc_interest_ms=&utm_loc_physical_ms=9067749&gclid=Cj0KCQjwh_eFBhDZARIsALHjIKdp8OdOA-1N8Fx5ZershN-VDobWwbz0fGulEfUoYTpjWtDX3NRO_mkaAmN9EALw_wcB) 참고해~

File: DatasetBuilder.py [hunk 3]
```diff
@@ -0,0 +1,87 @@
+import pandas as pd
+import numpy as np
+import random
+import torch
+
+from torch.utils.data import DataLoader
+from fieldTestDataset import FieldTestDataset
+
+def normalize_dataframes_to_nparrays(raw_dfs):
```
Comment (donghwan-shin) [inline DatasetBuilder.py:9]: 함수 파라미터랑 리턴 타입 힌트를 추가하고, 가능하면 함수 헤더 바로 아래에 간단한 설명도 추가하면 좋아. 예를 들어서, 

```python
def load_logs_into_df(log_format: str, log_files: list):
    """
    Load logs with parsing according to the given log format.
    :param log_format:
    :param log_files:
    :return:
    """
    header, pattern = generate_pattern_from_log_format(log_format)

    if 'message' not in header:
        print(f'ERROR: <message> is not in log_format={log_format}')
        exit(-1)
    ...
```

Comment (donghwan-shin) [reply DatasetBuilder.py:9]: 그리고 보니까 각 함수들이 잘 modularized 되어있는데, 이제 각 함수마다 normal/abnormal behavior 테스트 할 수 있는 unit tests 몇개씩 추가하면 완벽할 것 같아!

File: environment_model_generation.py (hunk 1 above)
Comment (yongjunshin) [reply environment_model_generation.py:83]: 네!! 찾아보니까, torch, numpy, random 등에서 각각만드 시드가 따로따로있는 것 같아서, 시드들을 통일하기 위해 랜덤실험을 할때도, 형 말대로 time으로 직접 랜덤시드를 만들고, 각 라이브러리마다 같은 값을 주도록할게요. 그리고, 그걸 반환하도록 추가할게요!

File: DatasetBuilder.py (hunk 3 above)
Comment (yongjunshin) [reply DatasetBuilder.py:9]: 네!! 함수별로 설명도 추가할게요!!
